drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/commands/ScenarioSimulationEventHandler.java

```diff
@@ -406,18 +407,17 @@ public void onEvent(SetPropertyHeaderEvent event) {
         context.getStatus().setPropertyNameElements(event.getPropertyNameElements());
         context.getStatus().setValueClassName(event.getValueClassName());
         context.getStatus().setImportPrefix(event.getImportPrefix());
-        if (isSameFactProperty(event.getGridWidget(), event.getPropertyNameElements()) &&
+        if (isSameFactProperty(event.getGridWidget(), event.getPropertyNameElements(), event.getFactMappingValueType()) &&
                 isSameSelectedColumnType(event.getGridWidget(), event.getValueClassName())) {
             return;
         } else {
             if (isSelectedColumnEmpty(event.getGridWidget())) {
                 commonExecution(new SetPropertyHeaderCommand(event.getGridWidget(), event.getFactMappingValueType()), true);
             } else {
-                if (!isSameFactProperty(event.getGridWidget(), event.getPropertyNameElements()) &&
-                        !isSameSelectedColumnType(event.getGridWidget(), event.getValueClassName())) {
-                    showDeletePopup(event);
-                } else {
+                if (isSameSelectedColumnType(event.getGridWidget(), event.getValueClassName()) && !ScenarioSimulationSharedUtils.isCollectionOrMap(event.getValueClassName())) {
```

Comment:
@jomarko Thank you, checking now.

Reply:
@jomarko I think it could be possible, I think it's better to don't explicitly bind that specific event as a method parameter, in order to promote its re-usage

Reply:
ok